geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geohash/GeohashUtils.scala

```diff
@@ -1065,7 +1065,7 @@ object GeohashUtils
       else Seq(ghMBR.toBinaryString.drop(minBits).take(usedBits)))
 
     // detect overflow
-    if (bitPrefixes.overflowed) return Seq()
+    if (bitPrefixes.overflowed) throw new IllegalStateException("Bit prefixes overflowed while calculating unique geohash substrings in polygon.")
```

Comment:
[picky?] 1) "geohash" -> "Geohash" (as per geohash.org); 2) it might be nice to dump the arguments to this call, so that the log contains enough information to reproduce the problem